cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -369,32 +370,64 @@ private String getSecurityProtocol(boolean tls, boolean sasl)   {
         if (authorization != null) {
             List<String> superUsers = new ArrayList<>();
-            String authorizerClass = "";
 
             // Broker super users
             superUsers.add(String.format("User:CN=%s,O=io.strimzi", KafkaResources.kafkaStatefulSetName(clusterName)));
             superUsers.add(String.format("User:CN=%s-%s,O=io.strimzi", clusterName, "entity-operator"));
             superUsers.add(String.format("User:CN=%s-%s,O=io.strimzi", clusterName, "kafka-exporter"));
 
-            // User configured super users
-            if (KafkaAuthorizationSimple.TYPE_SIMPLE.equals(authorization.getType())) {
-                KafkaAuthorizationSimple simpleAuthz = (KafkaAuthorizationSimple) authorization;
-                authorizerClass = KafkaAuthorizationSimple.AUTHORIZER_CLASS_NAME;
-
-                if (simpleAuthz.getSuperUsers() != null && simpleAuthz.getSuperUsers().size() > 0) {
-                    superUsers.addAll(simpleAuthz.getSuperUsers().stream().map(e -> String.format("User:%s", e)).collect(Collectors.toList()));
-                }
-            }
-
             printSectionHeader("Authorization");
-            writer.println("authorizer.class.name=" + authorizerClass);
+            configureAuthorization(clusterName, superUsers, authorization);
             writer.println("super.users=" + String.join(";", superUsers));
             writer.println();
         }
 
         return this;
     }
 
+    /**
+     * Configures authorization for the Kafka brokers. This method is used only internally.
+     *
+     * @param clusterName Name of the cluster
+     * @param superUsers Super users list who have all the rights on the cluster
+     * @param authorization The authorization configuration from the Kafka CR
+     */
+    private void configureAuthorization(String clusterName, List<String> superUsers, KafkaAuthorization authorization) {
+        if (KafkaAuthorizationSimple.TYPE_SIMPLE.equals(authorization.getType())) {
+            KafkaAuthorizationSimple simpleAuthz = (KafkaAuthorizationSimple) authorization;
+            writer.println("authorizer.class.name=" + KafkaAuthorizationSimple.AUTHORIZER_CLASS_NAME);
+
+            // User configured super users
+            if (simpleAuthz.getSuperUsers() != null && simpleAuthz.getSuperUsers().size() > 0) {
+                superUsers.addAll(simpleAuthz.getSuperUsers().stream().map(e -> String.format("User:%s", e)).collect(Collectors.toList()));
+            }
+        } else if (KafkaAuthorizationKeycloak.TYPE_KEYCLOAK.equals(authorization.getType())) {
+            KafkaAuthorizationKeycloak keycloakAuthz = (KafkaAuthorizationKeycloak) authorization;
+            writer.println("authorizer.class.name=" + KafkaAuthorizationKeycloak.AUTHORIZER_CLASS_NAME);
+            writer.println("principal.builder.class=" + KafkaAuthorizationKeycloak.PRINCIPAL_BUILDER_CLASS_NAME);
```

Comment:
Can you double check what impact does this have on the internal super users? Do they still keey their own names?

Reply:
They should keep their names, yes.

The JwtKafkaPrincipalBuilder only does something different when user was authenticated over SASL_OAUTHBEARER as seen [here](https://github.com/mstruk/strimzi-kafka-oauth/blob/authz/oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/JwtKafkaPrincipalBuilder.java). An internal user would be handled by the same logic as before.